rust/agents/relayer/src/msg/processor.rs

```diff
@@ -94,10 +97,18 @@ impl ProcessorExt for MessageProcessor {
 
             debug!(%msg, "Sending message to submitter");
 
+            let domain: HyperlaneDomain = KnownHyperlaneDomain::try_from(msg.destination)?.into();
+            let app_context_classifier = AppContextClassifier::new(
+                self.mailboxes.get(&domain).map(Clone::clone),
```

Comment:
Because the mailbox is only used by AppContextClassifier for getting the default ISM, and we don't care about ISMs, wondering if we should use something else than just use AppContextClassifier? that way we wouldn't need to add `mailboxes: HashMap<HyperlaneDomain, Arc<dyn Mailbox>>,`, as rn it's not actually used for anything

e.g. maybe rename `AppContextClassifier` to `IsmAwareAppContextClassifier` or something, and pull out the bit that we actually use for queue classification (that doesn't care about ISMs) into its own function that we use here